### PR TITLE
Warn if isometric TileMap is not Y-sorted

### LINKED ISSUE #57266: New TileMap editor: Tilemap creates kind of layereffect at the edge of the y-axis when painting a tile
### Godot version

4.0.alpha1.official

### System information

Windows 10 , Amd Ryzen5 3500u , Radeon Vega Mobile ,Vulkan Mobile

### Issue description

I have only one Layer in my Tileset but when i paint a Tile Godot draws a kind of layereffect.

https://user-images.githubusercontent.com/50340215/151217085-40ec0368-aec6-4015-9797-954c887354b2.mp4

.

### Steps to reproduce

Take one of the 3 Tiles and paint around of the y-axis edge.

### Minimal reproduction project

[Godot 4 Test.zip](https://github.com/godotengine/godot/files/7943943/Godot.4.Test.zip)

### LINKED ISSUE #65177: Isometric Tilemap Chunk Render Order
### Godot version

4.0 1c037c8

### System information

Windows10, AMD Radeon RX 5700 XT, Vulkan

### Issue description

In isometric tilemaps, the render order of the tiles are bugged.
The number of "incorrect" tiles in a row is in the following formula:

If even cell_quadrant_size: cell_quadrant_size / 2 - 1
If odd cell_quadrant_size: cell_quadrant_size  - 1
Example: if 16 then 16 / 2 - 1 = 7, if 7 then 7 - 1 = 6.

Illustration

### Steps to reproduce

Change the cell_quadrant_size of the tilemap.

### Minimal reproduction project

[IsoTilemap.zip](https://github.com/godotengine/godot/files/9467123/IsoTilemap.zip)

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes #57266

*Bugsquad edit: closes https://github.com/godotengine/godot/issues/65177*

### CODE REVIEW

File: scene/2d/tile_map.cpp [hunk 1]
```diff
@@ -3863,6 +3863,22 @@ PackedStringArray TileMap::get_configuration_warnings() const {
...
+			for (int layer = 0; layer < (int)layers.size(); layer++) {
+				if (!layers[layer].y_sort_enabled) {
+					warn = true;
+					break;
+				}
+			}
+		}
+
+		if (warn) {
+			warnings.push_back(RTR("Isometric TileSet works best if TileMap and all its layers have Y-sort enabled."));
```
Comment (MewPurPur) [inline scene/2d/tile_map.cpp:3878]: I feel like a warning should be more stern than telling you what's best.

How about "Isometric TileSet will likely not look as intended without Y-sort enabled for the TileMap and all of its layers?"

**Comment by MewPurPur** [comment]
> However the warning does not update when changing Y-sort of the TileMap. Not sure how to detect that.

Solving this would also help with the warning in #67192 :P

Can't we just do an `update_configuration_warnings()` call in `TileMap::set_y_sort_enabled(value)`?

**Review by groud — APPROVED** [review]

**Comment by groud** [comment]
I'd rather have not to need that warning, but I think, as the main use case would require it on, that it makes sense to have it.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**